test/alternator/pytest.ini

```diff
@@ -1,12 +0,0 @@
-# Pytest configuration file. If we don't have one in this directory,
-# pytest will look for one in our ancestor directories, and may find
-# something irrelevant - so we should keep this file even if in the future
-# we decide that we don't need to configure anything.
-
-[pytest]
-
-# Ignore warnings about HTTPS requests without certificate verification
-# (see issue #15287). Pytest breaks urllib3.disable_warnings() in conftest.py,
-# so we need to do this here.
-filterwarnings =
-    ignore::urllib3.exceptions.InsecureRequestWarning
```

Comment:
Adding pytest.ini makes it not one project, but subprojects. This will not allow running all tests with one command. For example, if there will be the same option declared, pytest will try to add it from each conftest and will throw an exception, since the option is already added.
Yes, it will take the in account tox.ini file. From my point of view, silencing warnings and cutting corners is not a good way of dealing with code issues.

Reply:
I agree with what you said in the first paragraph (I didn't say you were wrong to do these things - just that you should have mentioned it in the commit message).
But I don't understand the last paragraph - tox.ini is full of garbage that someone decided to add to the project 7 years ago and only causes stupid warnings when running pytest, so my testpy.ini avoided it - and yours probably too so everything is fine.

None of this is related to "silencing warnings". The specific "silencing warning" patch here that you moved from test/alternator/pytest.ini to test/pytest.ini - which is fine - was a stupid warning message from urllib3 that we can't fix in the code because it's a library, and because of pytest problems (which I explained in the comment) the only way to avoid these warnings is in pytest.ini. I think it's perfectly fine.

Reply:
tox.ini is not a garbage, but I suspect its overkill in our case. We can use pytest's pyproject.toml to run some tools for analysis, instead of tox.